.github/workflows/ci.yaml

```diff
@@ -136,3 +136,39 @@ jobs:
           labels: ${{ steps.meta.outputs.labels }}
           cache-from: type=gha
           cache-to: type=gha,mode=max
+
+      - name: Cache WASM filter
+        # Cached only on tag creation
+        if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+        id: cache-wasm-filter
+        uses: actions/cache@v3
+        with:
+          path: build/main.wasm
+          key: ${{ hashFiles('wasmplugin/**') }}
+
+  # Triggered only on tag creation
+  release:
+    if: github.event_name == 'push' && contains(github.ref, 'refs/tags/')
+    needs: build
+    name: Release
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - uses: actions/cache/restore@v3
+        id: cache-wasm-filter
+        with:
+          path: build/main.wasm
+          key: ${{ hashFiles('wasmplugin/**') }}
+          fail-on-cache-miss: true
+
+      - name: Create draft release
+        run: |
+          ls build
+          tag="${GITHUB_REF#refs/tags/}"
+          mv build/main.wasm build/coraza-proxy-wasm-${tag}.wasm
```

Comment:
Let's not rename the wasm file and instead zip it into a versioned archive and publish that instead of the direct wasm